File: src/main/java/rag/rerank/NoOpReranker.java
package rag.reranker;
import java.util.*;
import rag.retrieval.Hit;

public class NoOpReranker implements Reranker {
    
    // return wihtout doing any operation
    @Override
    public List<Hit> rerank(List<String> terms, List<Hit> hits, Object config) {
        return hits;
    }
}

File: src/main/python/rag/rerank/noop_reranker.py
from typing import List, Any
from rag.rerank.reranker import Reranker
from rag.retrieval.hit import Hit


class NoOpReranker(Reranker):
    def rerank(self, terms: List[str], hits: List[Hit], config: Any) -> List[Hit]:
        return hits






